src/key_value_store/core/KeyValueStore.cpp

```diff
@@ -0,0 +1,66 @@
+#include "KeyValueStore.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+// Alias for the unorderedList
+typedef std::unordered_map<std::string, 
+  std::vector<std::string>> hashMap; 
+
+bool KeyValueStore::Put(std::string key, std::string value) {
+  bool result; 
+  lock_.lock();
+  // If this key already exists in the store
+  if (kv_store_.find(key) != kv_store_.end()) {
+    // If user puts the same function
```

Comment:
I don't understand this comment

Reply:
I'm assuming that the all the functions are unique. If we are mapping event-type to functions, I am assuming that we would not want the event type to map to two of the same functions. i.e if event 1 maps to registerUser() I am assuming we don't want registerUser to be in the value vector twice. Is this assumption wrong / should I remove this check?

Reply:
Hmm, I'm still confused what function means here -- if this is the kvstore it has no knowledge of functions that provide any functionality because it just knows about keys and values.

Reply:
My bad, I misunderstood the functionality of this store. Have removed this part of the function, please check changed version.